src/java/htsjdk/samtools/metrics/MetricBase.java

```diff
@@ -84,7 +85,10 @@ public int hashCode() {
         int result = 0;
         for (final Field f : getClass().getFields()) {
             try {
-                result = 31 * result + f.get(this).hashCode();
+                Object value = f.get(this);
+                value = new FormatUtil().format(value); //format the value the way it will be written to disk
```

Comment:
@lbergelson I see what you're saying.  And I see the logic of your argument.  Something still bothers me about using String formatted values in equals() and hashCode().

It's definitely simpler than, say, rounding floating point values for comparison, or doing an "approximate equality" check on floating point numbers.  And you're not wrong that the equals() and hashCode() methods are probably not performance sensitive on these classes.

Can I suggest a couple of minor changes before merging:
1) On line 89 extract the call to new FormatUtil() and move it outside the loop
2) Add/correct the documentation on equals() and hashCode() to explain in context why they are comparing String formatted versions of things.